src/transaction.ts

```diff
@@ -31,15 +31,29 @@ export function buildTx(
   psetBase64: string,
   unspents: UtxoInterface[],
   recipients: RecipientInterface[],
-  changeScriptByAsset: ChangeAddressFromAssetGetter,
+  changeAddressByAsset: ChangeAddressFromAssetGetter,
   addFee: boolean = false,
   satsPerByte: number = 0.1,
   network: networks.Network = networks.regtest
 ): string {
+  if (satsPerByte < 0.1) {
+    throw new Error('satsPerByte minimum value is 0.1');
+  }
+
+  if (recipients.length === 0) {
+    throw new Error(
+      'need a least one recipient output to build the transaction'
+    );
+  }
+
+  if (unspents.length === 0) {
+    throw new Error('need at least one unspent to fund the transaction');
+  }
+
```

Comment:
Too simple checks IMO:

1. I would create a specific method where incapsulate all the validation logic.
2. satsPerByte MUST not be mandatory since the fee addition is optional. It would need to be checked only if addFee is not false.
3. recipients MUST not be mandatory since this method should allow to add inputs OR outputs (so ins or outs or both of them!). This applies also for the unspents.
4. if both the unspents and the recipients are provided, then we need to make sure that for every asset of the recipients there's at least one unspent of the same asset.

Reply:
I agree for 1, 2 & 3. However for 4 -> this check is costly in terms of computation resources (we need to crawl all the UTXOS and this array could be huge). Moreover the coin selection will also crawl over unspents and should handle this type of errors (so the function will failed anyway at the select utxos step, even if we do not validate the unspents array)

Reply:
For point 3 that's not really true: `buildTx` selects unspents among utxos set to fill the recipients requirements. Thus, both are linked. However, we could export `addToTx` function for simple add outputs/inputs use cases.

Reply:
Re: 3

We shouldnt be a general purposes function, otherwise it becomes a leaky abstraction IMHO with a lot of subtle things to document etc.. So Agrre with @louisinger  here